src/mono/mono/component/debugger.c

```diff
@@ -28,8 +28,11 @@ MonoComponentDebugger *
 mono_component_debugger_init (void)
 {
 	debugger_agent_add_function_pointers (&fn_table);
-#ifdef TARGET_WASM
+#if defined(TARGET_WASM) && !defined(HOST_WASI)
 	mini_wasm_debugger_add_function_pointers (&fn_table);
+#endif
+#if defined(HOST_WASI)
```

Comment:
Does this need to check !target_wasm too?

Reply:
Always when we have host_wasi we will have target_wasm = true.